lib/Common/Memory/Recycler.cpp

```diff
@@ -5559,8 +5577,22 @@ Recycler::WaitForConcurrentThread(DWORD waitTime)
         SetThreadPriority(this->concurrentThread, THREAD_PRIORITY_NORMAL);
     }
 
+#ifdef ENABLE_BASIC_TELEMETRY
+    bool isBlockingMainThread = GetCurrentThreadContextId() == mainThreadId;
```

Comment:
This seems like it should always be true rather than checking the threadId (which also has rental threading considerations i think)